runelite-client/src/main/java/net/runelite/client/plugins/PluginManager.java

```diff
@@ -210,6 +219,10 @@ public void startCorePlugins()
 	{
 		boolean developerPlugins = RuneLite.getOptions().has("developer-mode");
 
+		MutableGraph<Class<? extends Plugin>> graph = GraphBuilder
```

Comment:
I feel like this graph should be transformed to immutable one at the end of the instantiation and stopPlugin and startPlugin  should check for dependencies and call stopPlugin and startPlugin on them respectively as well, so entire "subtree" of the plugin would be effectively traversed and plugin will stop all it's dependencies as well. Maybe using something like https://github.com/google/guava/blob/master/guava/src/com/google/common/graph/Traverser.java#L63 should work? Not sure, I don't really know the graph package of guava.